src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -188,7 +187,7 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                         _success = result;
 
                         // Replay Warnings and Errors from an existing lock file
-                        await MSBuildRestoreUtility.ReplayWarningsAndErrorsAsync(_request.ExistingLockFile, _logger);
+                        await MSBuildRestoreUtility.ReplayWarningsAndErrorsAsync(_request.ExistingLockFile?.LogMessages, _logger);
```

Comment:
Why do we need the null check?

Reply:
The `ExistingLockFile` property is now lazy and could return null if there is no assets file

Reply:
Oh, the null check used to be in the replay warnings method.